include/fmt/format.h

```diff
@@ -1685,8 +1685,9 @@ template <typename Range> class basic_writer {
     handle_int_type_spec(spec.type, int_writer<T, Spec>(*this, value, spec));
   }
 
+#if FMT_USE_FLOAT || FMT_USE_DOUBLE || FMT_USE_LONG_DOUBLE
   template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value)>
-  void write(T value, format_specs specs = {}) {
+  void write_float(T value, format_specs specs = {}) {
```

Comment:
Instead of renaming and conditionally compiling this function, I suggest adding something like:

```c++
    if (const_check(
      (std::is_same<T, float>::value && !FMT_USE_FLOAT) ||
      (std::is_same<T, double>::value && !FMT_USE_DOUBLE) ||
      (std::is_same<T, long double>::value && !FMT_USE_LONG_DOUBLE))) {
      return;
    }
```

Reply:
Done.